### PR TITLE
[MIG] stock_picking_invoice_link: Migration to 11.0

### PR DESCRIPTION
**Pull request by aitorbouzas**
Migration to 11.0.
This command failed:
`git format-patch --keep-subject --stdout origin/11.0..origin/10.0 -- $MODULE | git am -3 --keep`
I guess that's why all the files seem to be added instead of modified...

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This is not correct. If fails, you have to fix the conflicts, but what you can't do is to dismiss original commit authorship.

**Comment by aitorbouzas** [comment]
Ok, I understand. This is the error I get:

```Applying: [add] stock_picking_invoice_link
Using index info to reconstruct a base tree...
.git/rebase-apply/patch:29: trailing whitespace.
#
.git/rebase-apply/patch:105: trailing whitespace.
#
.git/rebase-apply/patch:128: trailing whitespace.

.git/rebase-apply/patch:132: trailing whitespace.

.git/rebase-apply/patch:140: trailing whitespace.

warning: squelched 1 whitespace error
warning: 6 lines add whitespace errors.
Falling back to patching base and 3-way merge...
Auto-merging stock_picking_invoice_link/__init__.py
CONFLICT (add/add): Merge conflict in stock_picking_invoice_link/__init__.py
error: Failed to merge in the changes.
Patch failed at 0001 [add] stock_picking_invoice_link
The copy of the patch that failed is found in: .git/rebase-apply/patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
```
How can I solve this? Sorry for the inconvenience, i'm still learning git...

**Comment by pedrobaeza** [comment]
Solve the conflict editing the file and letting the correct code. Then, `git add --all` and `git am --continue`. This is not something Odoo related, but GIT related.

**Comment by aitorbouzas** [comment]
This thing is giving me lots of errors. Am I doing this right? I have a duplicated 11.0 branch of stock-logistics-workflow without the module stock_picking_invoice_link. Then I use that command and shows that error, solve the file and use the command again, then it shows another error, and so on, this is endless. I need a bit of help.

**Comment by pedrobaeza** [comment]
Well, sometimes happen that there are several iterations on this. That is usually because a merge operation has been done in the commit history. It's a pity and a dull job, but we have to deal with that until we fix it. Next migrations (v12, v13) won't have this problem as the procedure has been changed to this one.

**Comment by aitorbouzas** [comment]
Isn't there a way to tell the command that newer commits are the ones that should be applied? I'll try to deal with it if not.

**Comment by pedrobaeza** [comment]
I'm afraid not (or I don't know it), sorry.

**Comment by aitorbouzas** [comment]
I'll close this by now. If someone wants to do the job, i've got the module working in V11.

**Comment by pedrobaeza** [comment]
OK, as you want. Sometimes, it doesn't consist only in having it "working". You are losing visibility as contributor, losing CI tools for checking if everything is all right, etc, but OK, if you don't want to do the job, we can't force you.

**Comment by aitorbouzas** [comment]
It's not that I don't want, it's simply that I don't have enough time for it right now. Of course I would like to gain visibility, but i'm not able to do that job right now.

**Comment by pedrobaeza** [comment]
OK, if you want, we can do it together for this first time. This is all what I can offer to you. Think that not first time is the hardest one. The rest will be smoother.

**Comment by aitorbouzas** [comment]
Ok, tell me how I must proceed.

**Comment by pedrobaeza** [comment]
Next week I'm out of my office. But following one we can make a hangout to see this.

**Comment by aitorbouzas** [comment]
Ok, thank you.

**Comment by pedrobaeza** [comment]
OK, I'm back, so when do you want to see it?

**Comment by pedrobaeza** [comment]
This one is fixed and ready to be reviewed

**Comment by aitorbouzas** [comment]
Tests still need to be fixed!

**Comment by pedrobaeza** [comment]
Please try to adapt them

**Comment by celm1990** [comment]
Hello,
It is ready to merge or need help???, I can collaborate

**Comment by pedrobaeza** [comment]
@celm1990 you can make finish this PR fixing the tests.

**Comment by celm1990** [comment]
OK.
This fields can drop???
[invoice_line_id on stock_move](https://github.com/OCA/stock-logistics-workflow/pull/409/files#diff-631f3b718a7466e5e88e96aeb6838236R17)
[invoice_id on stock_picking](https://github.com/OCA/stock-logistics-workflow/pull/409/files#diff-6ce8bedefe3fb9c7ebae49b7f0e166e8R18)
Because fields m2m are better

File: stock_picking_invoice_link/tests/test_stock_picking_invoice_link.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        # deliver partially
+        self.assertEqual(self.so.invoice_status, 'no',
+                         'Sale Stock: so invoice_status should be '
+                         '"nothing to invoice" after invoicing')
+        pick_1 = self.so.picking_ids.filtered(
+            lambda x: x.picking_type_code == 'outgoing' and
+            x.state in ('confirmed', 'assigned', 'partially_available'))
+        pick_1.force_assign()
+        pick_1.move_line_ids.write({'qty_done': 1})
+        wiz_act = pick_1.do_new_transfer()
```
Comment (chienandalu) [inline stock_picking_invoice_link/tests/test_stock_picking_invoice_link.py:77]: You have to check how this is handled in v11 and change the test accordingly

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: stock_picking_invoice_link/tests/test_stock_picking_invoice_link.py (hunk 1 above)
Comment (aitorbouzas) [reply stock_picking_invoice_link/tests/test_stock_picking_invoice_link.py:77]: Yup, but @celm1990 made another PR... Should I continue with this?

**Comment by pedrobaeza** [comment]
We are going to continue the work, so I close this one (and probably the other). Thanks all for the efforts.

**Pull request closed without merging**